### PR TITLE
Add SafeArea for NavigationRail

### LINKED ISSUE #107407: NavigationRail is covered by notch, still not resovled
As mentioned in https://github.com/flutter/flutter/issues/102263 the NavigationRail is covered by the notch when an iPhone is in landscape, the solution to this would be to put the NavigationRail inside a SafeArea but that also makes the whole Content ignore the rounded corners which looks quite weird.

### PR DESCRIPTION
**Pull request by QuncCccccc**
This pr adds a SafeArea for NavigationRail component so that the NavigationRail will not be covered by notch/system navigation bar in landscape mode.

Screen Shot 2022-07-13 at 5 50 20 PM

Fixes #107407.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/navigation_rail.dart [hunk 1]
```diff
@@ -409,52 +409,55 @@ class _NavigationRailState extends State<NavigationRail> with TickerProviderStat
...
-                          onTap: () {
-                            if (widget.onDestinationSelected != null) {
-                              widget.onDestinationSelected!(i);
-                            }
-                          },
-                          indexLabel: localizations.tabLabel(
-                            tabIndex: i + 1,
-                            tabCount: widget.destinations.length,
+          child: SafeArea(
+            right: false,
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/navigation_rail.dart:413]: just double checking, on devices with RTL locale, the navigation rail is still on the left?

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

File: packages/flutter/lib/src/material/navigation_rail.dart (hunk 1 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/navigation_rail.dart:413]: No, it should be on the right side. Thanks so much for catching this!! Fixing!

**Comment by thilojaeggi** [comment]
The NavigationRail is now much wider even when not on the side of the notch. It's a solution, but not the prettiest one.
Is this intended?

**Pull request merged**